keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -173,42 +175,34 @@ Since images can be pre-loaded, loaded outside the `kubelet` process, and
 garbage collected.. the list of images that required authentication in `kubelet`
 will not be a source of truth for how all images were pulled that are in the
 container runtime cache. To mitigate, images can be garbage collected at boot.
-And we will persist ensure metadata across reboot of host, and restart
+And for alpha, we will not persist ensure metadata across reboot of host, and restart
 of kubelet, and possibly look at a way to add ensure metadata for images loaded
 outside of kubelet. In beta we will add a switch to enable re-auth on boot for
 admins seeking that instead of having to garbage collect where they do not use
 or expect preloaded images since boot.
 
-
 ## Design Details
 
-Kubelet will track, in memory, a hash map for the credentials that were successfully used to pull an image. It has been decided that the hash map will be persisted to disk, in alpha.
-
-The persisted "cache" will undergo cleanup operations on a timely basis (by default once an hour).
+Kubelet will track, in memory, a hash map for the credentials that were successfully
+used to pull an image.
 
-The persistence of the on storage cache is mainly for restarting kubelet and/or node reboot.
+See PR linked above for detailed design / behavior documentation.
 
-The max size of the cache will scale with the number of unique cache entries * the number of unique images that have not been garbage collected. It is not expected that this will be a significant number of bytes. Will be verified by actual use in Alpha and subsequent metrics in Beta.
+Kubelet will add a new flag, named `PullImageSecretRecheckDuration` to make
+the expired duration configurable. The default value could be 1d. For a pod with
+IfNotPresent image pull policy and an image pull secret, kubelet will recheck
+the secret after `PullImageSecretRecheckDuration`.
 
-See `/var/lib/kubelet/image_manager_state` in [kubernetes/kubernetes#114847](https://github.com/kubernetes/kubernetes/pull/114847)
+To make the cluster in most secure situation, set `PullImageSecretRecheckDuration` to 0,
```

Comment:
pull always already forces recheck every time..

Reply:
iow this duration is for pull if not present.. not pull always or pull never... 

so should not be seen as an override to anything but that policy .. if not present since last access was verified

Reply:
Yes. I add a description in `Not Goal` as well that this feature will not affect pods with `Never`/`Always` image pull policy.